mne/preprocessing/xdawn.py

```diff
@@ -241,6 +241,10 @@ def fit(self, epochs, y=None):
         self : Xdawn instance
             The Xdawn instance.
         """
+
+        # Ensure epochs order
+        epochs = epochs[np.argsort(epochs.events[:, 0])]
```

Comment:
just sort if not sorted already. Otherwise you make a copy for nothing.

something like

np.all(np.diff(epochs.events[:, 0]) >= 0)